spec/models/user_spec.rb

```diff
@@ -0,0 +1,71 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe User, type: :model do
+  let(:user) { build(:user) }
+  let(:user2) { build(:user) }
+  let(:user3) { build(:user) }
+
+  it 'creates valid user' do
+    expect(user.save).to be(true)
+  end
+
+  context 'is not saved because it' do
+    it 'forbides blank username' do
+      user = build(:user, username: '')
+      user.save
+      expect(user.errors[:username]).to include("can't be blank")
+    end
+
+    it 'requires valid email' do
+      user = build(:user, email: '')
+      user.save
+      expect(user.errors[:email]).to include("can't be blank")
+
+      user.email = 'wrongmicro.com'
+      user.save
+      expect(user.errors[:email]).to include('is invalid')
+    end
+
+    it 'requires password to contain at least 6 characters' do
+      user = build(:user, password: '')
+      user.save
+      expect(user.errors[:password]).to include("can't be blank")
+
+      user.password = 'admin'
+      user.save
+      expect(user.errors[:password]).to include('is too short (minimum is 6 characters)')
+    end
+  end
+
+  context 'friendships' do
+    before(:each) do
```

Comment:
We can likewise just use:

```
let(:friendship1) { create(:friendship, to_user: user2, from_user: user) }
let(:friendship2) { create(:friendship, to_user: user3, from_user: user) }
```
It isn't a dogma to prefer `let` over `before` but it's **generally** better to do it that way. Read more [here](https://dzone.com/articles/rspec-let-vs-before).